File: R/example1.R
#==================================================================================================
#                     Copyright (C) 2016 Olivier Mallet - All Rights Reserved                      
#==================================================================================================

suppressMessages(library(RcppURT))

run <- function()
{
  x = as.matrix(read.table("../data/x.csv", sep=","))
  y = as.matrix(read.table("../data/y.csv", sep=","))

  # running OLS regression as in ./examples/example1.cpp using double precision type
  fit = OLS_d$new(y, x, stats=TRUE)
  fit$show()

  # running OLS regression as in ./examples/example1.cpp using single precision type
  fit = OLS_f$new(y, x, stats=TRUE)
  fit$show()

  # running first ADF test as in ./examples/example2.cpp using double precision type
  test = ADF_d$new(y, lags=10, trend='ct')
  test$show()

  # running second ADF test as in ./examples/example2.cpp using double precision type
  test$method = 'AIC'
  test$bootstrap = TRUE
  test$niter = 10000
  test$show()
}

File: examples/example1.cpp
//=================================================================================================
//                    Copyright (C) 2016 Olivier Mallet - All Rights Reserved                      
//=================================================================================================

#include "../include/URT.hpp"

int main()
{
   int nrows = 1000;
   int ncols = 10;

   // generating random arrays
   urt::Vector<double> y = urt::wiener_process<double>(nrows);
   urt::Matrix<double> x = urt::wiener_process<double>(nrows, ncols);

   // adding intercept to matrix of independent variables
   urt::add_intercept(x);

   // writting data to CSV files
   urt::WriteToCSV("./URT/data/y.csv", y);
   urt::WriteToCSV("./URT/data/x.csv", x);

   // running OLS regression
   urt::OLS<double> fit(y, x, true);

   // outputting regression results
   fit.show();
}
